File: binarySearch/search_range.py
def searchRange(nums, target):
   l = 0
   r = len(nums) - 1
   while l <= r:
      mid = (l + r) // 2
      if nums[mid] == target:
         l = mid
         r = mid
         while nums[l - 1] == target:
            l = l - 1
         while nums[r + 1] == target:
            r = r + 1
         return [l, r]
      elif nums[mid] < target:
         l = mid + 1
      else:
         r = mid - 1
   return [-1, -1]

File: binarySearch/searchRange.js
const searchRange = function (nums, target) {
   let l = 0
   let r = nums.length - 1
   while (l <= r) {
      let mid = Math.floor((l + r) / 2)
      if (nums[mid] === target) {
         let l = mid
         let r = mid
         while (nums[l - 1] === target) {
            l--
         }
         while (nums[r + 1] === target) {
            r++
         }
         return [l, r]
      } else if (nums[mid] < target) {
         l = mid + 1
      } else {
         r = mid - 1
      }
   }
   return [-1, -1]
};